scipy/optimize/_lsq/least_squares.py

```diff
@@ -241,12 +242,13 @@ def loss_function(f, cost_only=False):
     return loss_function
 
 
+@_workers_wrapper
```

Comment:
Just for my own understanding as it's been a while since I used the workers, are we cleaning up the user's workers in case they want to do something else with the pool ? Not sure if that is a problem or not.

Reply:
If the user supplies their own workers then Mapwrapper doesn't do any cleanup.
It only cleans up any resources (Pools) that it creates.
The purpose of the decorator is so that we don't need to do all that wrangling inside least squares.

Reply:
The decorator is so that we don't need to setup Mapwrapper within least squares

Reply:
Ah my bad, I've read the decorator backwards. Then apologies for the noise.

Reply:
No, I appreciate the eyes. The main thing I'd like to know is about lmdif and lmder. At the moment `method='lm'` uses two point forward differences with lmdif. That's not immediately accessible to parallelise because it's done in C. I was wondering if we could change the lm method to use approx_derivative with lmder, and whether the two would be equivalent.

Reply:
This is what the authors wrote half a century ago 

![image](https://github.com/user-attachments/assets/05a25f40-10a4-45f4-b705-92962c3fd091)

Reply:
And this is the explanation of the Jacobian detail 

![image](https://github.com/user-attachments/assets/7226ae33-273d-436a-a97c-f353e1d57e18)

Reply:
As far as I can tell, they are equivalent, contingent upon the correctness of the Jacobian calculator. So I'd be willing to bet on yes you can do that. The authors have no reservations either way.